subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/TaskOutputFilesRepository.java

```diff
@@ -14,11 +14,12 @@
  * limitations under the License.
  */
 
-package org.gradle.internal.cleanup;
+package org.gradle.api.internal.changedetection.state;
 
 import java.io.File;
 
-public interface BuildOutputDeleter {
+public interface TaskOutputFilesRepository {
```

Comment:
I think we are not registering task outputs but the repository is here for querying and changing information about task outputs. So I would keep using `Repository`.